spec.md

```diff
@@ -5,76 +5,56 @@ keywords: registry, on-prem, images, tags, repository, distribution, api, advanc
-    - [DELETE Manifest](#delete-manifest)
-  - [Blob](#blob)
-    - [GET Blob](#get-blob)
-      - [Fetch Blob](#fetch-blob)
-      - [Fetch Blob Part](#fetch-blob-part)
-    - [DELETE Blob](#delete-blob)
-  - [Initiate Blob Upload](#initiate-blob-upload)
-    - [POST Initiate Blob Upload](#post-initiate-blob-upload)
-      - [Initiate Monolithic Blob Upload](#initiate-monolithic-blob-upload)
-      - [Initiate Resumable Blob Upload](#initiate-resumable-blob-upload)
-      - [Mount Blob](#mount-blob)
-  - [Blob Upload](#blob-upload)
-    - [GET Blob Upload](#get-blob-upload)
-    - [PATCH Blob Upload](#patch-blob-upload)
-    - [PUT Blob Upload](#put-blob-upload)
-    - [DELETE Blob Upload](#delete-blob-upload)
-
-## Notational Conventions
-
-The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" are to be interpreted as described in [RFC 2119](http://tools.ietf.org/html/rfc2119) (Bradner, S., "Key words for use in RFCs to Indicate Requirement Levels", BCP 14, RFC 2119, March 1997).
-
-The key words "unspecified", "undefined", and "implementation-defined" are to be interpreted as described in the [rationale for the C99 standard][c99-unspecified].
-
-An implementation is not compliant if it fails to satisfy one or more of the MUST, MUST NOT, REQUIRED, SHALL, or SHALL NOT requirements for the protocols it implements.
-An implementation is compliant if it satisfies all the MUST, MUST NOT, REQUIRED, SHALL, and SHALL NOT requirements for the protocols it implements.
-
-## Historical Context
+The Open Container Initiative Distribution Specification defines an API protocol to facilitate and standardize the distribution of content, especially related to container images and associated artifacts.
+
+### Historical Context
+
+The spec is based on the specification for the [Docker Registry HTTP API V2 protocol](https://github.com/docker/distribution/blob/5cb406d511b7b9163bff9b6439072e4892e5ae3b/docs/spec/api.md).
 
 For relevant details and a history leading up to this specification, please see the following issues:
 
 - [moby/moby#8093](https://github.com/moby/moby/issues/8093)
 - [moby/moby#9015](https://github.com/moby/moby/issues/9015)
 - [docker/docker-registry#612](https://github.com/docker/docker-registry/issues/612)
 
-<!--- TODO: add relevant background information here --->
+### Definitions
+
+Several terms are used frequently in this document and warrant basic definitions:
+
+- **Registry**: a HTTP service which implements this spec
+- **Client**: a tool that communicates with registries over HTTP
+- **Push**: the act of uploading content to a registry
+- **Pull**: the act of downloading content from a registry
+- **Artifact**: a single piece of content, made up of a manifest and one or more layers
+- **Manifest**: a JSON document which defines an artifact
+- **Layer**: a single part of all the parts which comprise an artifact
```

Comment:
Layer explanation from image spec: https://github.com/opencontainers/image-spec/blob/master/config.md#oci-image-configuration

suggest: ```**Layer**: a set of filesystem changes; Layers are typically used by a layer-based or union filesystem to form the mountable root file system of a container.``` and maybe link to the oci spec

Reply:
I think we should avoid using the term layers except in the context of images. I wish we would have just renamed layers to references. I don't think we need to inject that confusion here.

Reply:
fair.. in the context of images image layers are a set of..

Reply:
I don't think the image spec is useful in this context. I'd rather define Layer as legacy terminology referring to blobs that are specifically for container images.

Reply:
what is "this context?" ... Why is OCI image not useful in the context of the words used in the OCI distribution spec? Let's say an OCI image is but one type of OCI artifact... Most definitions include examples, for example, in the context of OCI container images, layers are a set of ...  IMO the inclusion of examples does not make a definition as confusing as the lack of examples does. Esp. when highjacking a term that is common in the vernacular.